### DISCUSSION TITLE
nlist parameter limitations

### DISCUSSION THREAD

**Discussion by EngineerXL** [milvus-io/milvus#34457] [Q&A and General discussion]
I loaded 10M 512D dataset into FAISS IVF_PQ and achieved good accuracy/RAM/speed with m=128, nbits=8, nlist=262144, nprobe = 8192
Now I'm interested in how to set up a similar index with Milvus
I've read the doc, and the question is why is the nlist parameter limited up to 65536?

**Comment by xiaofan-luan** [comment]
it doesn't make sense to set that much nlist.

usually set to 4096 or 16384 would be good enough.

Because milvus doesn't build index on all data, for each 100w data(500MB Segment szie), 4096 is good enough, if your segment has 500w data(2GB Segment size), 16384 is good enough
